### ISSUE TITLE
python 3.11 CACHE opcode

### ISSUE THREAD

**Issue by AdrienPensart** [pulkin/unethical-python-hacks#1]
hello!

I would like to experiment your code with python 3.11 but they introduced CACHE opcode at code beginning, so the code no longer works.
And they replaced CALL_FUNCTION with CALL opcode...

The idea is to implement a kind of "?" operator like in rust to replace that code : 
```
def foo():
    return None:

if foo() is None:
    return None
```
by
```
def bar():
    early_return_if_none(foo())
```

or even : 
```
@early_return
def foo():
    return None

def bar():
    foo() # early return there
```

Is there a way to increment pos to find the CALL and patch it at this place ?

I tried but my code/knowledge is not good enough.

**Comment by pulkin** [comment]
Yeah there are some massive changes of the bytecode language in 3.11. First, many opcodes are followed-up by caches of different sizes so something harcoded in flow_control no longer works. Second, CALLs are now done in two opcodes. But if you address these two it should be fine. I will be doing this for https://github.com/pulkin/pyteleport
